### PR TITLE
image: fix handling of var/lib/extrausers when preseeding uc20

### PR DESCRIPTION
**Pull request by stolowski**
Create and mount `var/lib/extrausers` under writable. Fixes https://bugs.launchpad.net/snapd/+bug/1978850

Note, to test the fix you need to:
- use `snap prepare-image ...` command built from this branch
- seed a custom snapd snap with the updated `data/preseed.json` (i.e. re-package snapd from edge and replace preseed.json inside).

### CODE REVIEW

File: tests/main/preseed-core20/task.yaml [hunk 1]
```diff
@@ -51,7 +53,7 @@ execute: |
   SNAPD_SNAP=$(ls /tmp/snapd*.snap)
 
   # XXX: confusingly, the name of our test key is actually " (test)".
-  snap prepare-image --preseed --preseed-sign-key=" (test)" --channel=stable --snap="$SNAPD_SNAP" "$TESTSLIB"/assertions/developer1-20-dangerous.model "$PREPARE_IMAGE_DIR" > preseed.log 2>&1
+  snap prepare-image --preseed --preseed-sign-key=" (test)" --channel=stable --snap="$SNAPD_SNAP" --snap systemusers-snap_1.0_all.snap "$TESTSLIB"/assertions/developer1-20-dangerous.model "$PREPARE_IMAGE_DIR" > preseed.log 2>&1
```
Comment (mvo5) [inline tests/main/preseed-core20/task.yaml:56]: Thanks for adding this!

**Review by mvo5 — APPROVED** [review]
\o/ Thank you!

File: image/preseed/preseed_linux.go [hunk 2]
```diff
@@ -291,7 +291,7 @@ func prepareCore20Mountpoints(prepareImageDir, tmpPreseedChrootDir, snapdSnapBlo
 	for _, dir := range []string{
 		"etc/udev/rules.d", "etc/systemd/system", "etc/dbus-1/session.d",
 		"var/lib/snapd/seed", "var/cache/snapd", "var/cache/apparmor",
-		"var/snap", "snap"} {
+		"var/snap", "snap", "var/lib/extrausers"} {
```
Comment (bboozzoo) [inline image/preseed/preseed_linux.go:294]: nitpick about formatting:
```suggestion
		"var/snap", "snap", "var/lib/extrausers",
	} {
```

File: tests/lib/prepare.sh [hunk 3]
```diff
@@ -444,6 +444,10 @@ repack_snapd_snap_with_deb_content_and_run_mode_firstboot_tweaks() {
     local UNPACK_DIR="/tmp/snapd-unpack"
     unsquashfs -no-progress -d "$UNPACK_DIR" snapd_*.snap
 
+    # data/preseed.json is not included in the deb, use the latest
+    # version from source tree to replace the one in the re-packed snapd snap.
+    cp /home/gopath/src/github.com/snapcore/snapd/data/preseed.json "$UNPACK_DIR"/usr/lib/snapd
```
Comment (bboozzoo) [inline tests/lib/prepare.sh:449]: I think this should work;
```suggestion
    cp "$PROJECT_PATH/data/preseed.json" "$UNPACK_DIR"/usr/lib/snapd
```

**Review by bboozzoo — APPROVED** [review]
Some nitpicks, otherwise LGTM

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Comment by kubiko** [comment]
LGTM
than you @stolowski

**Review by kubiko — APPROVED** [review]
LGTM
Thank you @stolowski !

**Pull request merged**